### ISSUE TITLE
[BUG]: FileNotFoundError: [WinError 2] The system cannot find the file specified [bentoml-deployer demo] 

### ISSUE THREAD

**Issue by Yuanlim0919** [zenml-io/zenml#1595]
### System Information

ZENML_LOCAL_VERSION: 0.38.0
ZENML_SERVER_VERSION: 0.38.0
ZENML_SERVER_DATABASE: mysql
ZENML_SERVER_DEPLOYMENT_TYPE: other
ZENML_CONFIG_DIR: C:\Users\acer\AppData\Roaming\zenml
ZENML_LOCAL_STORE_DIR: C:\Users\acer\AppData\Roaming\zenml\local_stores
ZENML_SERVER_URL: http://localhost:8080
ZENML_ACTIVE_REPOSITORY_ROOT: C:\Users\acer
PYTHON_VERSION: 3.10.11
ENVIRONMENT: native
SYSTEM_INFO: {'os': 'windows', 'windows_version_release': '10', 'windows_version': '10.0.19045',
'windows_version_service_pack': 'SP0', 'windows_version_os_type': 'Multiprocessor Free'}
ACTIVE_WORKSPACE: default
ACTIVE_STACK: stack_test
ACTIVE_USER: admin
TELEMETRY_STATUS: enabled
ANALYTICS_CLIENT_ID: 20d4e795-cb65-407f-b728-45875558b544
ANALYTICS_USER_ID: ff84ca8e-83c4-4cd4-bcd4-f2992d4b3147
ANALYTICS_SERVER_ID: a679abec-f1d4-4296-9aed-9203abbe644f
INTEGRATIONS: ['airflow', 'aws', 'bentoml', 'graphviz', 'kaniko', 'pillow', 'pytorch', 's3', 'scipy', 'sklearn']
PACKAGES: {'pip': '22.2', 'babel': '2.12.1', 'configupdater': '3.1.1', 'deprecated': '1.2.13', 'flask': '2.2.5',
'flask-appbuilder': '4.1.4', 'flask-babel': '2.0.0', 'flask-caching': '2.0.2', 'flask-jwt-extended': '4.5.2',
'flask-login': '0.6.2', 'flask-sqlalchemy': '2.5.1', 'flask-wtf': '1.1.1', 'gitpython': '3.1.31', 'jinja2': '3.1.2',
'mako': '1.2.4', 'markdown': '3.4.3', 'markupsafe': '2.1.2', 'pillow': '9.5.0', 'pyjwt': '2.7.0', 'pymysql': '1.0.3',
'pyyaml': '5.4.1', 'pygments': '2.15.1', 'sqlalchemy': '1.4.41', 'sqlalchemy-jsonfield': '1.0.1.post0',
'sqlalchemy-utils': '0.38.3', 'send2trash': '1.8.2', 'wtforms': '3.0.1', 'werkzeug': '2.2.3', 'aiobotocore': '2.4.2',
'aiohttp': '3.8.4', 'aioitertools': '0.11.0', 'aiosignal': '1.3.1', 'alembic': '1.8.1', 'analytics-python':
'1.4.post1', 'anyio': '3.6.2', 'apache-airflow': '2.4.3', 'apache-airflow-providers-common-sql': '1.5.1',
'apache-airflow-providers-ftp': '3.4.1', 'apache-airflow-providers-http': '4.4.1', 'apache-airflow-providers-imap':
'3.2.1', 'apache-airflow-providers-sqlite': '3.4.1', 'apispec': '3.3.2', 'appdirs': '1.4.4', 'argcomplete': '3.0.8',
'argon2-cffi': '21.3.0', 'argon2-cffi-bindings': '21.2.0', 'arrow': '1.2.3', 'asgiref': '3.7.1', 'asttokens': '2.2.1',
'async-timeout': '4.0.2', 'attrs': '22.2.0', 'backcall': '0.2.0', 'backoff': '1.10.0', 'bcrypt': '4.0.1',
'beautifulsoup4': '4.12.2', 'bentoml': '1.0.20', 'bleach': '6.0.0', 'blinker': '1.6.2', 'boto3': '1.26.144',
'botocore': '1.27.59', 'build': '0.10.0', 'cachelib': '0.9.0', 'cattrs': '22.2.0', 'certifi': '2023.5.7', 'cffi':
'1.15.1', 'charset-normalizer': '3.1.0', 'circus': '0.18.0', 'click': '8.1.3', 'click-option-group': '0.5.5',
'click-params': '0.3.0', 'clickclick': '20.10.2', 'cloudpickle': '2.2.1', 'colorama': '0.4.6', 'colorlog': '4.8.0',
'comm': '0.1.3', 'commonmark': '0.9.1', 'connexion': '2.14.2', 'contextlib2': '21.6.0', 'contourpy': '1.0.7',
'cron-descriptor': '1.4.0', 'croniter': '1.3.15', 'cryptography': '41.0.1', 'cycler': '0.11.0', 'databricks-cli':
'0.17.7', 'debugpy': '1.6.7', 'decorator': '5.1.1', 'deepmerge': '1.1.0', 'defusedxml': '0.7.1', 'dill': '0.3.6',
'distro': '1.8.0', 'dnspython': '2.3.0', 'docker': '6.0.1', 'docutils': '0.20.1', 'ecdsa': '0.18.0',
'email-validator': '1.3.1', 'entrypoints': '0.4', 'exceptiongroup': '1.1.1', 'executing': '1.2.0', 'fastapi':
'0.75.2', 'fastapi-utils': '0.2.1', 'fastjsonschema': '2.16.3', 'filelock': '3.12.0', 'flask-session': '0.5.0',
'fonttools': '4.39.4', 'fqdn': '1.5.1', 'frozenlist': '1.3.3', 'fs': '2.4.16', 'fsspec': '2022.11.0', 'gitdb':
'4.0.10', 'google-pasta': '0.2.0', 'graphviz': '0.20.1', 'greenlet': '2.0.2', 'grpcio': '1.54.2',
'grpcio-health-checking': '1.54.2', 'gunicorn': '20.1.0', 'h11': '0.14.0', 'httpcore': '0.17.2', 'httplib2': '0.19.1',
'httptools': '0.5.0', 'httpx': '0.24.1', 'idna': '3.4', 'importlib-metadata': '4.13.0', 'inflection': '0.5.1',
'ipykernel': '6.23.0', 'ipython': '8.13.2', 'ipython-genutils': '0.2.0', 'ipywidgets': '7.7.5', 'isoduration':
'20.11.0', 'itsdangerous': '2.1.2', 'jedi': '0.18.2', 'jmespath': '1.0.1', 'joblib': '1.2.0', 'jsonpointer': '2.3',
'jsonschema': '4.17.3', 'jupyter-client': '8.2.0', 'jupyter-core': '5.3.0', 'jupyter-events': '0.6.3',
'jupyter-server': '2.5.0', 'jupyter-server-terminals': '0.4.4', 'jupyterlab-pygments': '0.2.2', 'jupyterlab-widgets':
'1.1.4', 'kiwisolver': '1.4.4', 'lazy-object-proxy': '1.9.0', 'linkify-it-py': '2.0.2', 'lockfile': '0.12.2',
'markdown-it-py': '2.2.0', 'marshmallow': '3.19.0', 'marshmallow-enum': '1.5.1', 'marshmallow-oneofschema': '3.0.1',
'marshmallow-sqlalchemy': '0.26.1', 'matplotlib': '3.7.1', 'matplotlib-inline': '0.1.6', 'mdit-py-plugins': '0.3.5',
'mdurl': '0.1.2', 'mistune': '2.0.5', 'mlflow': '2.3.1', 'monotonic': '1.6', 'mpmath': '1.3.0', 'multidict': '6.0.4',
'multiprocess': '0.70.14', 'nbclassic': '1.0.0', 'nbclient': '0.7.4', 'nbconvert': '7.4.0', 'nbformat': '5.8.0',
'nest-asyncio': '1.5.6', 'networkx': '3.1', 'notebook': '6.5.4', 'notebook-shim': '0.2.3', 'numpy': '1.24.3',
'oauthlib': '3.2.2', 'opentelemetry-api': '1.14.0', 'opentelemetry-instrumentation': '0.35b0',
'opentelemetry-instrumentation-aiohttp-client': '0.35b0', 'opentelemetry-instrumentation-asgi': '0.35b0',
'opentelemetry-instrumentation-grpc': '0.35b0', 'opentelemetry-sdk': '1.14.0', 'opentelemetry-semantic-conventions':
'0.35b0', 'opentelemetry-util-http': '0.35b0', 'orjson': '3.8.14', 'packaging': '23.1', 'pandas': '2.0.1',
'pandocfilters': '1.5.0', 'parso': '0.8.3', 'passlib': '1.7.4', 'pathos': '0.3.0', 'pathspec': '0.9.0', 'pendulum':
'2.1.2', 'pickleshare': '0.7.5', 'pip-requirements-parser': '32.0.1', 'pip-tools': '6.13.0', 'platformdirs': '3.5.0',
'pluggy': '1.0.0', 'pox': '0.3.2', 'ppft': '1.7.6.6', 'prison': '0.2.1', 'prometheus-client': '0.16.0',
'prompt-toolkit': '3.0.38', 'protobuf': '3.20.3', 'protobuf3-to-dict': '0.1.5', 'psutil': '5.9.5', 'pure-eval':
'0.2.2', 'pyarrow': '11.0.0', 'pyasn1': '0.5.0', 'pycparser': '2.21', 'pydantic': '1.10.7', 'pyngrok': '6.0.0',
'pynvml': '11.5.0', 'pyparsing': '2.4.7', 'pyproject-hooks': '1.0.0', 'pyrsistent': '0.19.3', 'python-daemon':
'3.0.1', 'python-dateutil': '2.8.2', 'python-dotenv': '1.0.0', 'python-jose': '3.3.0', 'python-json-logger': '2.0.7',
'python-multipart': '0.0.6', 'python-nvd3': '0.15.0', 'python-slugify': '8.0.1', 'python-terraform': '0.10.1', 'pytz':
'2023.3', 'pytzdata': '2020.1', 'pywin32': '306', 'pywinpty': '2.0.10', 'pyzmq': '25.0.2', 'querystring-parser':
'1.2.4', 'requests': '2.30.0', 'requests-toolbelt': '1.0.0', 'rfc3339-validator': '0.1.4', 'rfc3986-validator':
'0.1.1', 'rich': '12.6.0', 'rsa': '4.9', 's3fs': '2022.11.0', 's3transfer': '0.6.1', 'sagemaker': '2.117.0', 'schema':
'0.7.5', 'scikit-learn': '1.2.2', 'scipy': '1.10.1', 'setproctitle': '1.3.2', 'setuptools': '66.0.0', 'simple-di':
'0.1.5', 'six': '1.16.0', 'smdebug-rulesconfig': '1.0.1', 'smmap': '5.0.0', 'sniffio': '1.3.0', 'soupsieve': '2.4.1',
'sqlalchemy2-stubs': '0.0.2a34', 'sqlmodel': '0.0.8', 'sqlparse': '0.4.4', 'stack-data': '0.6.2', 'starlette':
'0.17.1', 'swagger-ui-bundle': '0.0.9', 'sympy': '1.12', 'tabulate': '0.9.0', 'tenacity': '8.2.2', 'termcolor':
'2.3.0', 'terminado': '0.17.1', 'text-unidecode': '1.3', 'threadpoolctl': '3.1.0', 'tinycss2': '1.2.1', 'tomli':
'2.0.1', 'torch': '2.0.1', 'torchvision': '0.15.2', 'tornado': '6.3.1', 'traitlets': '5.9.0', 'typing-extensions':
'4.5.0', 'tzdata': '2023.3', 'uc-micro-py': '1.0.2', 'unicodecsv': '0.14.1', 'uri-template': '1.2.0', 'urllib3':
'1.26.15', 'uvicorn': '0.17.6', 'validators': '0.18.2', 'waitress': '2.1.2', 'watchfiles': '0.19.0', 'watchgod':
'0.8.2', 'wcwidth': '0.2.6', 'webcolors': '1.13', 'webencodings': '0.5.1', 'websocket-client': '1.5.1', 'websockets':
'11.0.3', 'wheel': '0.38.4', 'widgetsnbextension': '3.6.4', 'wrapt': '1.15.0', 'yarl': '1.9.2', 'zenml': '0.38.0',
'zipp': '3.15.0'}
←[33mThe stack ←[0m←[33mstack_test←[33m contains components that require building Docker images. Older versions of ZenML always built these images locally, but since version 0.32.0 this behavior can be configured using the ←[0m←[33mimage_builder←[33m stack component. This stack will temporarily default to a local image builder that mirrors the previous behavior, but this will be removed in future versions of ZenML. Please add an image builder to this stack:
←[0m←[33mzenml image-builder register <NAME> ...
zenml stack update 0ac3a99e-9f44-4917-8a2c-163ffede74c7 -i <NAME>←[33m←[0m

CURRENT STACK

Name: stack_test
ID: 0ac3a99e-9f44-4917-8a2c-163ffede74c7
Shared: No
User: admin / ff84ca8e-83c4-4cd4-bcd4-f2992d4b3147
Workspace: default / e6d3ad08-426e-4cbb-a367-fd816184cb7f

ORCHESTRATOR: default

Name: default
ID: 7a5cbef0-12ee-4532-be7f-3471b65383fe
Type: orchestrator
Flavor: local
Configuration: {}
Shared: No
User: admin / ff84ca8e-83c4-4cd4-bcd4-f2992d4b3147
Workspace: default / e6d3ad08-426e-4cbb-a367-fd816184cb7f

ARTIFACT_STORE: default

Name: default
ID: dd539621-c548-4563-9f86-23e333d7affa
Type: artifact_store
Flavor: local
Configuration: {'path': ''}
Shared: No
User: admin / ff84ca8e-83c4-4cd4-bcd4-f2992d4b3147
Workspace: default / e6d3ad08-426e-4cbb-a367-fd816184cb7f

MODEL_DEPLOYER: bentoml

Name: bentoml
ID: a7a1bb0a-bd8d-49f4-bb33-2bbb3be9778d
Type: model_deployer
Flavor: bentoml
Configuration: {'service_path': ''}
Shared: No
User: admin / ff84ca8e-83c4-4cd4-bcd4-f2992d4b3147
Workspace: default / e6d3ad08-426e-4cbb-a367-fd816184cb7f

EXPERIMENT_TRACKER: mlflow

Name: mlflow
ID: 280d3e6b-b94e-4f12-977e-c846f8471098
Type: experiment_tracker
Flavor: mlflow
Configuration: {'experiment_name': None, 'nested': False, 'tags': {}, 'tracking_uri': None, 'tracking_username':
'********', 'tracking_password': '********', 'tracking_token': '********', 'tracking_insecure_tls': False,
'databricks_host': None}
Shared: No
User: admin / ff84ca8e-83c4-4cd4-bcd4-f2992d4b3147
Workspace: default / e6d3ad08-426e-4cbb-a367-fd816184cb7f

### What happened?

I pulled the [bentoml-deployment] example repository into my local desktop, and I created a stack with default orchestrator, default artifact-store and bentoml deployer. When I execute ``run.py``, the execution of data loading step and training step is normal, when it comes ``bento_builder`` step, following error occurs:

 FileNotFoundError: [WinError 2] The system cannot find the file specified

### Relevant log output

```shell
Failed to run step bento_builder.
Pipeline run training_fashion_mnist-2023_06_05-09_25_28_938428 failed.
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\runpy.py:196 in _run_module_as_main                       │
│                                                                                                  │
│   193 │   main_globals = sys.modules["__main__"].__dict__                                        │
│   194 │   if alter_argv:                                                                         │
│   195 │   │   sys.argv[0] = mod_spec.origin                                                      │
│ ❱ 196 │   return _run_code(code, main_globals, None,                                             │
│   197 │   │   │   │   │    "__main__", mod_spec)                                                 │
│   198                                                                                            │
│   199 def run_module(mod_name, init_globals=None,                                                │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\runpy.py:86 in _run_code                                  │
│                                                                                                  │
│    83 │   │   │   │   │      __loader__ = loader,                                                │
│    84 │   │   │   │   │      __package__ = pkg_name,                                             │
│    85 │   │   │   │   │      __spec__ = mod_spec)                                                │
│ ❱  86 │   exec(code, run_globals)                                                                │
│    87 │   return run_globals                                                                     │
│    88                                                                                            │
│    89 def _run_module_code(code, init_globals=None,                                              │
│                                                                                                  │
│ c:\Users\acer\.vscode\extensions\ms-python.python-2023.8.0\pythonFiles\lib\python\debugpy\__main │
│ __.py:39 in <module>                                                                             │
│                                                                                                  │
│   36 │                                                                                           │
│   37 │   from debugpy.server import cli                                                          │
│   38 │                                                                                           │
│ ❱ 39 │   cli.main()                                                                              │
│   40                                                                                             │
│                                                                                                  │
│ c:\Users\acer\.vscode\extensions\ms-python.python-2023.8.0\pythonFiles\lib\python\debugpy/..\deb │
│ ugpy\server\cli.py:430 in main                                                                   │
│                                                                                                  │
│   427 │   │   │   "code": run_code,                                                              │
│   428 │   │   │   "pid": attach_to_pid,                                                          │
│   429 │   │   }[options.target_kind]                                                             │
│ ❱ 430 │   │   run()                                                                              │
│   431 │   except SystemExit as exc:                                                              │
│   432 │   │   log.reraise_exception(                                                             │
│   433 │   │   │   "Debuggee exited via SystemExit: {0!r}", exc.code, level="debug"               │
│                                                                                                  │
│ c:\Users\acer\.vscode\extensions\ms-python.python-2023.8.0\pythonFiles\lib\python\debugpy/..\deb │
│ ugpy\server\cli.py:284 in run_file                                                               │
│                                                                                                  │
│   281 │   log.describe_environment("Pre-launch environment:")                                    │
│   282 │                                                                                          │
│   283 │   log.info("Running file {0!r}", target)                                                 │
│ ❱ 284 │   runpy.run_path(target, run_name="__main__")                                            │
│   285                                                                                            │
│   286                                                                                            │
│   287 def run_module():                                                                          │
│                                                                                                  │
│ c:\Users\acer\.vscode\extensions\ms-python.python-2023.8.0\pythonFiles\lib\python\debugpy\_vendo │
│ red\pydevd\_pydevd_bundle\pydevd_runpy.py:321 in run_path                                        │
│                                                                                                  │
│   318 │   │   # Not a valid sys.path entry, so run the code directly                             │
│   319 │   │   # execfile() doesn't help as we want to allow compiled files                       │
│   320 │   │   code, fname = _get_code_from_file(run_name, path_name)                             │
│ ❱ 321 │   │   return _run_module_code(code, init_globals, run_name,                              │
│   322 │   │   │   │   │   │   │   │   pkg_name=pkg_name, script_name=fname)                      │
│   323 │   else:                                                                                  │
│   324 │   │   # Finder is defined for path, so add it to                                         │
│                                                                                                  │
│ c:\Users\acer\.vscode\extensions\ms-python.python-2023.8.0\pythonFiles\lib\python\debugpy\_vendo │
│ red\pydevd\_pydevd_bundle\pydevd_runpy.py:135 in _run_module_code                                │
│                                                                                                  │
│   132 │   fname = script_name if mod_spec is None else mod_spec.origin                           │
│   133 │   with _TempModule(mod_name) as temp_module, _ModifiedArgv0(fname):                      │
│   134 │   │   mod_globals = temp_module.module.__dict__                                          │
│ ❱ 135 │   │   _run_code(code, mod_globals, init_globals,                                         │
│   136 │   │   │   │     mod_name, mod_spec, pkg_name, script_name)                               │
│   137 │   # Copy the globals of the temporary module, as they                                    │
│   138 │   # may be cleared when the temporary module goes away                                   │
│                                                                                                  │
│ c:\Users\acer\.vscode\extensions\ms-python.python-2023.8.0\pythonFiles\lib\python\debugpy\_vendo │
│ red\pydevd\_pydevd_bundle\pydevd_runpy.py:124 in _run_code                                       │
│                                                                                                  │
│   121 │   │   │   │   │      __loader__=loader,                                                  │
│   122 │   │   │   │   │      __package__=pkg_name,                                               │
│   123 │   │   │   │   │      __spec__=mod_spec)                                                  │
│ ❱ 124 │   exec(code, run_globals)                                                                │
│   125 │   return run_globals                                                                     │
│   126                                                                                            │
│   127                                                                                            │
│                                                                                                  │
│ d:\Internship@Webcomm\ieee-fraud-detection\zenml_examples\bentoml_deployment\run.py:85 in        │
│ <module>                                                                                         │
│                                                                                                  │
│   82                                                                                             │
│   83                                                                                             │
│   84 if __name__ == "__main__":                                                                  │
│ ❱ 85 │   main()                                                                                  │
│   86                                                                                             │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\site-packages\click\core.py:1130 in __call__              │
│                                                                                                  │
│   1127 │                                                                                         │
│   1128 │   def __call__(self, *args: t.Any, **kwargs: t.Any) -> t.Any:                           │
│   1129 │   │   """Alias for :meth:`main`."""                                                     │
│ ❱ 1130 │   │   return self.main(*args, **kwargs)                                                 │
│   1131                                                                                           │
│   1132                                                                                           │
│   1133 class Command(BaseCommand):                                                               │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\site-packages\click\core.py:1055 in main                  │
│                                                                                                  │
│   1052 │   │   try:                                                                              │
│   1053 │   │   │   try:                                                                          │
│   1054 │   │   │   │   with self.make_context(prog_name, args, **extra) as ctx:                  │
│ ❱ 1055 │   │   │   │   │   rv = self.invoke(ctx)                                                 │
│   1056 │   │   │   │   │   if not standalone_mode:                                               │
│   1057 │   │   │   │   │   │   return rv                                                         │
│   1058 │   │   │   │   │   # it's not safe to `ctx.exit(rv)` here!                               │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\site-packages\click\core.py:1404 in invoke                │
│                                                                                                  │
│   1401 │   │   │   echo(style(message, fg="red"), err=True)                                      │
│   1402 │   │                                                                                     │
│   1403 │   │   if self.callback is not None:                                                     │
│ ❱ 1404 │   │   │   return ctx.invoke(self.callback, **ctx.params)                                │
│   1405 │                                                                                         │
│   1406 │   def shell_complete(self, ctx: Context, incomplete: str) -> t.List["CompletionItem"]:  │
│   1407 │   │   """Return a list of completions for the incomplete value. Looks                   │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\site-packages\click\core.py:760 in invoke                 │
│                                                                                                  │
│    757 │   │                                                                                     │
│    758 │   │   with augment_usage_errors(__self):                                                │
│    759 │   │   │   with ctx:                                                                     │
│ ❱  760 │   │   │   │   return __callback(*args, **kwargs)                                        │
│    761 │                                                                                         │
│    762 │   def forward(                                                                          │
│    763 │   │   __self, __cmd: "Command", *args: t.Any, **kwargs: t.Any  # noqa: B902             │
│                                                                                                  │
│ d:\Internship@Webcomm\ieee-fraud-detection\zenml_examples\bentoml_deployment\run.py:69 in main   │
│                                                                                                  │
│   66 │   │   │   ),                                                                              │
│   67 │   │   │   bento_builder=bento_builder,                                                    │
│   68 │   │   │   deployer=bentoml_model_deployer,                                                │
│ ❱ 69 │   │   ).run()                                                                             │
│   70 │   if predict:                                                                             │
│   71 │   │   inference_fashion_mnist(                                                            │
│   72 │   │   │   inference_loader=inference_loader(),                                            │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\site-packages\zenml\pipelines\base_pipeline.py:602 in run │
│                                                                                                  │
│    599 │   │   │   # unexpected behavior                                                         │
│    600 │   │   │   constants.SHOULD_PREVENT_PIPELINE_EXECUTION = True                            │
│    601 │   │   │   try:                                                                          │
│ ❱  602 │   │   │   │   stack.deploy_pipeline(deployment=deployment_model)                        │
│    603 │   │   │   finally:                                                                      │
│    604 │   │   │   │   constants.SHOULD_PREVENT_PIPELINE_EXECUTION = False                       │
│    605                                                                                           │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\site-packages\zenml\stack\stack.py:864 in deploy_pipeline │
│                                                                                                  │
│    861 │   │   Returns:                                                                          │
│    862 │   │   │   The return value of the call to `orchestrator.run_pipeline(...)`.             │
│    863 │   │   """                                                                               │
│ ❱  864 │   │   return self.orchestrator.run(deployment=deployment, stack=self)                   │
│    865 │                                                                                         │
│    866 │   def _get_active_components_for_step(                                                  │
│    867 │   │   self, step_config: "StepConfiguration"                                            │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\site-packages\zenml\orchestrators\base_orchestrator.py:16 │
│ 6 in run                                                                                         │
│                                                                                                  │
│   163 │   │                                                                                      │
│   164 │   │   environment = get_config_environment_vars()                                        │
│   165 │   │   try:                                                                               │
│ ❱ 166 │   │   │   result = self.prepare_or_run_pipeline(                                         │
│   167 │   │   │   │   deployment=deployment, stack=stack, environment=environment                │
│   168 │   │   │   )                                                                              │
│   169 │   │   finally:                                                                           │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\site-packages\zenml\orchestrators\local\local_orchestrato │
│ r.py:82 in prepare_or_run_pipeline                                                               │
│                                                                                                  │
│    79 │   │   │   │   │   step.config.name,                                                      │
│    80 │   │   │   │   )                                                                          │
│    81 │   │   │                                                                                  │
│ ❱  82 │   │   │   self.run_step(                                                                 │
│    83 │   │   │   │   step=step,                                                                 │
│    84 │   │   │   )                                                                              │
│    85                                                                                            │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\site-packages\zenml\orchestrators\base_orchestrator.py:18 │
│ 6 in run_step                                                                                    │
│                                                                                                  │
│   183 │   │   │   step=step,                                                                     │
│   184 │   │   │   orchestrator_run_id=self.get_orchestrator_run_id(),                            │
│   185 │   │   )                                                                                  │
│ ❱ 186 │   │   launcher.launch()                                                                  │
│   187 │                                                                                          │
│   188 │   @staticmethod                                                                          │
│   189 │   def requires_resources_in_orchestration_environment(                                   │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\site-packages\zenml\orchestrators\step_launcher.py:198 in │
│ launch                                                                                           │
│                                                                                                  │
│   195 │   │   │                                                                                  │
│   196 │   │   │   if execution_needed:                                                           │
│   197 │   │   │   │   try:                                                                       │
│ ❱ 198 │   │   │   │   │   self._run_step(                                                        │
│   199 │   │   │   │   │   │   pipeline_run=pipeline_run,                                         │
│   200 │   │   │   │   │   │   step_run=step_run_response,                                        │
│   201 │   │   │   │   │   )                                                                      │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\site-packages\zenml\orchestrators\step_launcher.py:380 in │
│ _run_step                                                                                        │
│                                                                                                  │
│   377 │   │   │   │   │   step_run_info=step_run_info,                                           │
│   378 │   │   │   │   )                                                                          │
│   379 │   │   │   else:                                                                          │
│ ❱ 380 │   │   │   │   self._run_step_without_step_operator(                                      │
│   381 │   │   │   │   │   step_run_info=step_run_info,                                           │
│   382 │   │   │   │   │   input_artifacts=step_run.input_artifacts,                              │
│   383 │   │   │   │   │   output_artifact_uris=output_artifact_uris,                             │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\site-packages\zenml\orchestrators\step_launcher.py:450 in │
│ _run_step_without_step_operator                                                                  │
│                                                                                                  │
│   447 │   │   │   output_artifact_uris: The output artifact URIs of the current step.            │
│   448 │   │   """                                                                                │
│   449 │   │   runner = StepRunner(step=self._step, stack=self._stack)                            │
│ ❱ 450 │   │   runner.run(                                                                        │
│   451 │   │   │   input_artifacts=input_artifacts,                                               │
│   452 │   │   │   output_artifact_uris=output_artifact_uris,                                     │
│   453 │   │   │   step_run_info=step_run_info,                                                   │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\site-packages\zenml\orchestrators\step_runner.py:133 in   │
│ run                                                                                              │
│                                                                                                  │
│   130 │   │   │   self._stack.prepare_step_run(info=step_run_info)                               │
│   131 │   │   │   step_failed = False                                                            │
│   132 │   │   │   try:                                                                           │
│ ❱ 133 │   │   │   │   return_values = step_entrypoint(**function_params)                         │
│   134 │   │   │   except BaseException as step_exception:  # noqa: E722                          │
│   135 │   │   │   │   step_failed = True                                                         │
│   136 │   │   │   │   failure_hook_source = self.configuration.failure_hook_source               │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\site-packages\zenml\integrations\bentoml\steps\bento_buil │
│ der.py:103 in bento_builder_step                                                                 │
│                                                                                                  │
│   100 │   │   )                                                                                  │
│   101 │                                                                                          │
│   102 │   # Build the BentoML bundle                                                             │
│ ❱ 103 │   bento = bentos.build(                                                                  │
│   104 │   │   service=params.service,                                                            │
│   105 │   │   version=params.version,                                                            │
│   106 │   │   labels=labels,                                                                     │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\site-packages\simple_di\__init__.py:139 in _              │
│                                                                                                  │
│   136 │   │   bind = sig.bind_partial(*filtered_args, **filtered_kwargs)                         │
│   137 │   │   bind.apply_defaults()                                                              │
│   138 │   │                                                                                      │
│ ❱ 139 │   │   return func(*_inject_args(bind.args), **_inject_kwargs(bind.kwargs))               │
│   140 │                                                                                          │
│   141 │   setattr(_, "_is_injected", True)                                                       │
│   142 │   return cast(WrappedCallable, _)                                                        │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\site-packages\bentoml\bentos.py:372 in build              │
│                                                                                                  │
│   369 │   │   bentofile_path = os.path.join(os.path.dirname(f.name), f.name)                     │
│   370 │   │   build_args.extend(["--bentofile", bentofile_path])                                 │
│   371 │   │   try:                                                                               │
│ ❱ 372 │   │   │   output = subprocess.check_output(build_args)                                   │
│   373 │   │   except subprocess.CalledProcessError as e:                                         │
│   374 │   │   │   logger.error("Failed to build BentoService bundle: %s", e)                     │
│   375 │   │   │   raise                                                                          │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\subprocess.py:421 in check_output                         │
│                                                                                                  │
│    418 │   │   │   empty = b''                                                                   │
│    419 │   │   kwargs['input'] = empty                                                           │
│    420 │                                                                                         │
│ ❱  421 │   return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,                      │
│    422 │   │   │      **kwargs).stdout                                                           │
│    423                                                                                           │
│    424                                                                                           │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\subprocess.py:503 in run                                  │
│                                                                                                  │
│    500 │   │   kwargs['stdout'] = PIPE                                                           │
│    501 │   │   kwargs['stderr'] = PIPE                                                           │
│    502 │                                                                                         │
│ ❱  503 │   with Popen(*popenargs, **kwargs) as process:                                          │
│    504 │   │   try:                                                                              │
│    505 │   │   │   stdout, stderr = process.communicate(input, timeout=timeout)                  │
│    506 │   │   except TimeoutExpired as exc:                                                     │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\subprocess.py:971 in __init__                             │
│                                                                                                  │
│    968 │   │   │   │   │   self.stderr = io.TextIOWrapper(self.stderr,                           │
│    969 │   │   │   │   │   │   │   encoding=encoding, errors=errors)                             │
│    970 │   │   │                                                                                 │
│ ❱  971 │   │   │   self._execute_child(args, executable, preexec_fn, close_fds,                  │
│    972 │   │   │   │   │   │   │   │   pass_fds, cwd, env,                                       │
│    973 │   │   │   │   │   │   │   │   startupinfo, creationflags, shell,                        │
│    974 │   │   │   │   │   │   │   │   p2cread, p2cwrite,                                        │
│                                                                                                  │
│ C:\Users\acer\Anaconda3\envs\MLOps\lib\subprocess.py:1456 in _execute_child                      │
│                                                                                                  │
│   1453 │   │   │                                                                                 │
│   1454 │   │   │   # Start the process                                                           │
│   1455 │   │   │   try:                                                                          │
│ ❱ 1456 │   │   │   │   hp, ht, pid, tid = _winapi.CreateProcess(executable, args,                │
│   1457 │   │   │   │   │   │   │   │   │   │    # no special security                            │
│   1458 │   │   │   │   │   │   │   │   │   │    None, None,                                      │
│   1459 │   │   │   │   │   │   │   │   │   │    int(not close_fds),                              │
│                                                                                                  │
│ c:\Users\acer\.vscode\extensions\ms-python.python-2023.8.0\pythonFiles\lib\python\debugpy\_vendo │
│ red\pydevd\_pydev_bundle\pydev_monkey.py:901 in new_CreateProcess                                │
│                                                                                                  │
│    898 │   │   │   cmd_line = patch_arg_str_win(cmd_line)                                        │
│    899 │   │   │   send_process_created_message()                                                │
│    900 │   │                                                                                     │
│ ❱  901 │   │   return getattr(_subprocess, original_name)(app_name, cmd_line, *args)             │
│    902 │                                                                                         │
│    903 │   return new_CreateProcess                                                              │
│    904                                                                                           │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
FileNotFoundError: [WinError 2] The system cannot find the file specified
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by htahir1** [comment]
Maybe this is tangentially related to https://github.com/zenml-io/zenml/issues/1592.. We are investigating BentoML and will take this into account

**Comment by strickvl** [comment]
Thanks for the bug report. I've investigated this and we've figured out what's going wrong. I'll update once this has been fixed.

**Comment by strickvl** [comment]
@Yuanlim0919 this has been fixed in #1647 and was subsequently released. The fix will be in our most current version, i.e.. Thank you for bringing it to our attention!
